### PR TITLE
Add template for epic

### PR DESCRIPTION
**Pull request by maayanf24**
Add GitHub Issue template for a new epic, based on
the Definition of Done

Signed-off-by: Maayan Friedman <maafried@redhat.com>

### CODE REVIEW

**Review by Jaanki — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/epic.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+Why is this needed:
+
+## Acceptance Criteria
+<!-- Document acceptance criteria here: -->
+
+
+## Work Items
+<!-- Add work items here, 
+make sure you have an item to cover all relevant topics from DoD checklist: -->
```
Comment (tpantelis) [inline .github/ISSUE_TEMPLATE/epic.md]: Perhaps we should move this section after.

```suggestion
## Work Items
<!-- Add work items here. 
Make sure you have item(s) to cover all relevant topics from the checklist above: -->
```

File: .github/ISSUE_TEMPLATE/epic.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+name: Epic
+about: Add an epic issue to the Submariner project
+labels: epic
+
+---
+<!-- Please only use this template for submitting epic requests -->
+
+## EPIC Description
```
Comment (tpantelis) [inline .github/ISSUE_TEMPLATE/epic.md]:
```suggestion
## Epic Description
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/epic.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+---
+name: Epic
+about: Add an epic issue to the Submariner project
```
Comment (nyechiel) [inline .github/ISSUE_TEMPLATE/epic.md:3]:
```suggestion
about: Add an epic issue template to the Submariner project
```

**Review by aswinsuryan — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/epic.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+labels: epic
+
+---
+<!-- Please only use this template for submitting epic requests -->
```
Comment (nyechiel) [inline .github/ISSUE_TEMPLATE/epic.md]: Can you add a description of what is an epic? Something like: "Epics contain features or other work items that help deliver on a specific initiative. Epics are a helpful way to identify and organize the work."

File: .github/ISSUE_TEMPLATE/epic.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+Why is this needed:
+
+## Acceptance Criteria
+<!-- Document acceptance criteria here: -->
+
+## Definition of Done (Checklist)
+<!-- Make sure to check all relevant items before end of the release: -->
+
+* [ ] Code complete
+* [ ] The acceptance criteria met
```
Comment (nyechiel) [inline .github/ISSUE_TEMPLATE/epic.md:23]: What are the acceptance criteria?

File: .github/ISSUE_TEMPLATE/epic.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* [ ] Unit/e2e test added & pass
+* [ ] CI jobs pass
+* [ ] Deployed using cloud-prepare+subctl
```
Comment (nyechiel) [inline .github/ISSUE_TEMPLATE/epic.md:26]: What about Helm?

File: .github/ISSUE_TEMPLATE/epic.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+* [ ] Deployed using ACM/OCM addon
+* [ ] Deployed on supported platforms (for e.g kind, OCP on AWS, OCP on GCP)
+* [ ] Run subctl verify, diagnose and gather
+* [ ] Uninstall
+* [ ] Troubleshooting (gather/diagnose) added
+* [ ] Documentation added
+* [ ] Release notes added
+
```
Comment (nyechiel) [inline .github/ISSUE_TEMPLATE/epic.md:35]: Metrics?

**Review by nyechiel — COMMENTED** [review]
Thanks for putting this together, @maayanf24. I added a few questions/suggestions inline. 

One additional thing that I think worth discussing (and was brought by @sridhargaddam, @vthapar and @mkolesnik in the past) - is where do we want Epics to live? i.e is there a consensus that all epics should go under the enhancements repo only (and therefore this template should only be available there)?

File: .github/ISSUE_TEMPLATE/epic.md (hunk 3 above)
Comment (mkolesnik) [reply .github/ISSUE_TEMPLATE/epic.md:3]: Your suggestion doesn't make sense :)

This is the template, the text is what you'll see on GH.

File: .github/ISSUE_TEMPLATE/epic.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+name: Epic
+about: Add an epic issue to the Submariner project
+labels: epic
+
+---
+<!-- Please only use this template for submitting epic requests -->
+
+## Epic Description
+
+Why is this needed:
```
Comment (mkolesnik) [inline .github/ISSUE_TEMPLATE/epic.md]:
```suggestion
<!-- Describe why is this work needed and what needs to be done -->
```

File: .github/ISSUE_TEMPLATE/epic.md (hunk 5 above)
Comment (mkolesnik) [reply .github/ISSUE_TEMPLATE/epic.md:23]: I think this refers to the block in L13

File: .github/ISSUE_TEMPLATE/epic.md (hunk 3 above)
Comment (nyechiel) [reply .github/ISSUE_TEMPLATE/epic.md:3]: Oh right, please ignore :)

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by nyechiel — APPROVED** [review]

**Comment by nyechiel** [comment]
> Thanks for putting this together, @maayanf24. I added a few questions/suggestions inline.
> 
> One additional thing that I think worth discussing (and was brought by @sridhargaddam, @vthapar and @mkolesnik in the past) - is where do we want Epics to live? i.e is there a consensus that all epics should go under the enhancements repo only (and therefore this template should only be available there)?

@maayanf24 any thoughts on this?

**Comment by sridhargaddam** [comment]
> > Thanks for putting this together, @maayanf24. I added a few questions/suggestions inline.
> > One additional thing that I think worth discussing (and was brought by @sridhargaddam, @vthapar and @mkolesnik in the past) - is where do we want Epics to live? i.e is there a consensus that all epics should go under the enhancements repo only (and therefore this template should only be available there)?
> 
> @maayanf24 any thoughts on this?

I vote for having all the epics in the enhancements repo. It will make it easy to locate the EPICs which otherwise can be spread across different components and could be hard to trace while searching in the future.

**Comment by sridhargaddam** [comment]
As discussed over the slack, I'm merging this PR so that people can start using the template for the EPICs. If there are any further comments, let's push an additional PR to make changes.

**Pull request merged**